### PR TITLE
Add type hints for settings module

### PR DESCRIPTION
**Pull request by bjoernricks**
Types make it easier to understand the code and improve autocompletion in IDEs.
	modified:   pelican/utils.py#

# Pull Request Checklist

- [x] Ensured **tests pass** and (if applicable) updated functional test output
- [x] Conformed to **code style guidelines** by running appropriate linting tools

While reading pelican's code I just added some type hints. If accepted I can upload more PRs for additional type hint.

### CODE REVIEW

File: pelican/settings.py [hunk 1]
```diff
@@ -221,7 +227,7 @@ def getabs(maybe_relative, base_path=path):
     return settings
 
 
-def get_settings_from_module(module=None):
+def get_settings_from_module(module: Optional[object] = None) -> Settings:
```
Comment (avaris) [inline pelican/settings.py:230]: Why not `Optional[ModuleType]`?

Comment (bjoernricks) [reply pelican/settings.py]: Can't remember. I changed it.

**Review by avaris — APPROVED** [review]

**Comment by bjoernricks** [comment]
@avaris could you approve the workflow to be run. I guess that's missing for merging the PR.

**Review by justinmayer — APPROVED** [review]
Many thanks to @bjoernricks for the enhancement and to @avaris for reviewing ✨

**Pull request merged**